### PR TITLE
Update vendor to use pkg test

### PR DESCRIPTION
**Pull request by chizhg**
**What this PR does, why we need it**:
Some libs have been moved to pkg to avoid circular dependencies, update test-infra to vendor them.

### CODE REVIEW

File: pkg/testgrid/testgrid_test.go [hunk 1]
```diff
@@ -56,7 +57,7 @@ func TestXMLOutput(t *testing.T) {
 `
 
 	// Create a test file
-	if err := CreateXMLOutput(tc, name); err != nil {
+	if err := pkgTestgrid.CreateXMLOutput(tc, name); err != nil {
```
Comment (chaodaiG) [inline pkg/testgrid/testgrid_test.go:60]: This test should like in pkg repo

**Comment by chaodaiG** [comment]
I don't understand why testgrid coverage is 0 now

File: pkg/testgrid/testgrid_test.go (hunk 1 above)
Comment (chizhg) [reply pkg/testgrid/testgrid_test.go]: It's already in https://github.com/knative/pkg/blob/master/test/testgrid/testgrid_test.go, I just forgot to delete it here.

Deleted.

**Comment by chizhg** [comment]
> I don't understand why testgrid coverage is 0 now

It's because the current test is for `testgrid/yaml.go` but not `testgrid/testgrid.go`

**Comment by chaodaiG** [comment]
> > I don't understand why testgrid coverage is 0 now
> 
> It's because the current test is for `testgrid/yaml.go` but not `testgrid/testgrid.go`

Should it be renamed to `yaml_test.go`?

**Comment by chizhg** [comment]
> > > I don't understand why testgrid coverage is 0 now
> > 
> > 
> > It's because the current test is for `testgrid/yaml.go` but not `testgrid/testgrid.go`
> 
> Should it be renamed to `yaml_test.go`?

Ah yes, done.

**Pull request merged**